code/modules/surgery/bones.dm

```diff
@@ -154,3 +156,48 @@
 	var/datum/organ/external/affected = target.get_organ(target_zone)
 	user.visible_message("<span class='warning'>[user]'s hand slips, smearing [tool] in the incision in [target]'s [affected.display_name]!</span>" , \
 	"<span class='warning'>Your hand slips, smearing [tool] in the incision in [target]'s [affected.display_name]!</span>")
+
+
+
+//////BONE MENDER/////////
+/datum/surgery_step/bone_mender
+	allowed_tools = list(
+		/obj/item/weapon/bonesetter/bone_mender = 100,
+		)
+
+	priority = 0.1 //so it trys to do this before /glue_bone
+
+	min_duration = 80
+	max_duration = 90
+
+/datum/surgery_step/bone_mender/can_use(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
+	var/datum/organ/external/affected = target.get_organ(target_zone)
+	return (affected.open >= 2 || (target.species.anatomy_flags & NO_SKIN)) && affected.stage <= 5
+
+/datum/surgery_step/bone_mender/begin_step(mob/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
+	var/datum/organ/external/affected = target.get_organ(target_zone)
+	if (affected.stage <= 5)
+		user.visible_message("[user] starts grasping the damaged bone edges in [target]'s [affected.display_name] with \the [tool]." , \
+		"You start grasping the bone edges and fusing them in [target]'s [affected.display_name] with \the [tool].")
+	target.custom_pain("Something in your [affected.display_name] is causing you a lot of pain!", 1)
+	..()
+
+/datum/surgery_step/bone_mender/end_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
+	var/datum/organ/external/affected = target.get_organ(target_zone)
+	user.visible_message("<span class='notice'>[user] fuses [target]'s [affected.display_name] bone with \the [tool].</span>"  , \
+		"<span class='notice'>You fuse the bone in [target]'s [affected.display_name] with \the [tool].</span>" )
+	affected.stage = 1
```

Comment:
These assignments to `affected.stage` seem to be useless, I suppose you should only keep the final `= 0`